spec/concurrent/edge/future_spec.rb

```diff
@@ -171,6 +171,12 @@
       expect([queue.pop, queue.pop, queue.pop, queue.pop].sort).to eq [:async, :async, :sync, :sync]
     end
 
+    it 'supports setting timeout while waiting' do
+      future = Concurrent.future { sleep 1 }
```

Comment:
Another way to handle this more deterministically would be with `CountDownLatch`:

```ruby
start_latch = Concurrent::CountDownLatch.new
end_latch = Concurrent::CountDownLatch.new

future = Concurrent.future do
  start_latch.count_down
  end_latch.wait(1)
end

start_latch.wait(1)
future.wait(0.1)
expect(future).not_to be_completed
end_latch.count_down
```